### PR TITLE
Added EtwEventSourceTarget - EventSource with configurable ProviderName

### PR DESCRIPTION
**Pull request by snakefoot**
Making NLog.Etw into something that is useful (Instead of just being a proof-of-concept-project).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,22 +22,23 @@ Add the assembly and new target to NLog.config:
...
-      <!-- extensions is not needed in NLog 4+ -->
-      <extensions>
+    <!-- extensions is not needed in NLog 4+ -->
+    <extensions>
         <add assembly="NLog.Etw" />
-      </extensions>
+    </extensions>
 
-      <targets async="true">
-        <target xsi:type="ExtendedEventTracing"
```
Comment (304NotModified) [inline README.md:36]: isn't better to keep both examples?

Comment (snakefoot) [reply README.md:36]: No one wants to use the other targets. They force you to use the same ETW-ProviderName.

The idea is that you should have a unique name for each application.

Comment (snakefoot) [reply README.md:36]: Actually curious why no one have implemented `NLogEtwExtendedTarget`-constructor that takes EventSource as parameter. Then one could atleast use it from code.

Comment (304NotModified) [reply README.md:36]: OK, so maybe drop that target in the next major version?

**Comment by 304NotModified** [comment]
renamed the PR, is it correct?

File: README.md (hunk 1 above)
Comment (snakefoot) [reply README.md:36]: Will try and modify it so one can provide a custom EventSource, that implements an interface that takes NLog.LogEventInfo and has an EventSource-property-getter, so the target can query if enabled.

**Comment by snakefoot** [comment]
Rename looks good.

File: README.md (hunk 1 above)
Comment (304NotModified) [reply README.md:36]: That's for the future, isn't? I can merge and release this one?

Comment (snakefoot) [reply README.md:36]: I have a PR ready now. When this has been merged. But now the AppVeyor-queue is long because of your double-build-commits.

Comment (snakefoot) [reply README.md:36]: But yes that idea is for the "future". Not related to this PR.

**Comment by snakefoot** [comment]
@304NotModified Possible to have it merged? Like to create one more PR

**Comment by 304NotModified** [comment]
done!

**Pull request merged**